webapp/src/js/services/android-api.js

```diff
@@ -138,8 +139,21 @@ angular.module('inboxServices').factory('AndroidApi',
           // If we're viewing a tab, but not the primary tab, go to primary tab
           var primaryTab = $('.header .tabs').find('> a:visible:first');
           if (!primaryTab.is('.selected')) {
-            $state.go(primaryTab.attr('ui-sref'));
-            return true;
+            var uiSref = primaryTab.attr('ui-sref');
+            if (uiSref) {
+              $state.go(primaryTab.attr('ui-sref'));
+              return true;
+            } else {
+              var message = 'Attempt to back to an undefined state';  
+              return Feedback.submit(message, false, function(err) {
+                if (err) {
+                  $log.error('Error saving feedback', err);
+                  return false;
+                }
+
+                return true;
```

Comment:
I think this should return `false` so android handles the back click. Otherwise we could get stuck in a loop where the back button never works.